packages/mdx-deck-code-surfer/src/deck-code-surfer.js

```diff
@@ -1,93 +1,76 @@
 import CodeSurfer from "code-surfer";
 import React from "react";
-import { modes, incStep, decStep } from "mdx-deck";
-import { withDeck } from "mdx-deck/dist/context";
-import { withSlide } from "mdx-deck/dist/Slide";
+import { withDeck, withSlide } from "mdx-deck";
+import { setSteps } from "mdx-deck/dist/updaters";
+import { modes } from "mdx-deck/dist/constants";
 
 export default withDeck(
-  withSlide(
-    class extends React.Component {
-      componentDidMount() {
-        document.body.addEventListener("keydown", this.handleKeyDown);
-      }
+  class InnerCodeSurfer extends React.Component {
+    constructor(props) {
+      console.log("ctr");
+      super(props);
+      const { update, index } = props.deck;
+      const steps = props.steps ? props.steps.length : 0;
+      update(setSteps(index, steps));
+    }
 
-      componentWillUnmount() {
-        document.body.removeEventListener("keydown", this.handleKeyDown);
-      }
+    shouldComponentUpdate() {
```

Comment:
Not 100% sure the intent here, but should this be using the `nextProps` argument for checking?

Reply:
You are right, thanks!